Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -211,6 +228,47 @@ std::map<std::string, std::string> FindSXPeaks::validateInputs() {
     validationOutput["XResolution"] = "XResolution must be set to a value greater than 0";
   }
 
+  const int minNSpectraPerPeak = static_cast<int>(getProperty("MinNSpectraPerPeak"));
+  const int maxNSpectraPerPeak = static_cast<int>(getProperty("MaxNSpectraPerPeak"));
```

Comment:
I'm not sure we need the `static_cast`'s to ints, as long as the type on the left of the variable name is identical to the type of the property that is defined in the algorithm. In this case, the type casting will only work if you use a non-const int on the left of `minNSpectraPerPeak` like so
```suggestion
  int maxNSpectraPerPeak = getProperty("MaxNSpectraPerPeak");
```

Reply:
Good spot - thanks @robertapplin!

Reply:
Thanks Rob for noting this! I have removed the static_casts and kept the const on the rhs on few other places as well where there is no need to casts since the types are same on both sides